test/framework/kogitoapp.go

```diff
@@ -41,37 +41,44 @@ const (
 
 // KogitoAppDeployment defines the elements for the deployment of a kogito application
 type KogitoAppDeployment struct {
-	AppName     string
-	ContextDir  string
-	Runtime     v1alpha1.RuntimeType
-	Native      bool
-	Persistence bool
-	Events      bool
-	Labels      map[string]string
+	AppName            string
+	GitSourceURI       string
+	GitSourceReference string
+	ContextDir         string
+	Runtime            v1alpha1.RuntimeType
+	Native             bool
+	Persistence        bool
+	Events             bool
+	Labels             map[string]string
 }
 
-// DeployExample deploy a Kogito example
-func DeployExample(namespace string, installerType InstallerType, kogitoAppDeployment KogitoAppDeployment) error {
+// DeployService deploy a Kogito service
+func DeployService(namespace string, installerType InstallerType, kogitoAppDeployment KogitoAppDeployment) error {
 	GetLogger(namespace).Infof("%s deploy %s example %s with name %s, native %v, persistence %v, events %v and labels %v", installerType, kogitoAppDeployment.Runtime, kogitoAppDeployment.ContextDir, kogitoAppDeployment.AppName, kogitoAppDeployment.Native, kogitoAppDeployment.Persistence, kogitoAppDeployment.Events, kogitoAppDeployment.Labels)
 	switch installerType {
 	case CLIInstallerType:
-		return cliDeployExample(namespace, kogitoAppDeployment)
+		return cliDeployService(namespace, kogitoAppDeployment)
 	case CRInstallerType:
-		return crDeployExample(namespace, kogitoAppDeployment)
+		return crDeployService(namespace, kogitoAppDeployment)
 	default:
 		panic(fmt.Errorf("Unknown installer type %s", installerType))
 	}
 }
 
-// DeployExample deploys an example
-func crDeployExample(namespace string, kogitoAppDeployment KogitoAppDeployment) error {
+func crDeployService(namespace string, kogitoAppDeployment KogitoAppDeployment) error {
 	kogitoApp := getKogitoAppStub(namespace, kogitoAppDeployment.AppName, kogitoAppDeployment.Labels)
 	kogitoApp.Spec.Runtime = kogitoAppDeployment.Runtime
 
 	kogitoApp.Spec.Build.Native = kogitoAppDeployment.Native
-	kogitoApp.Spec.Build.GitSource.URI = config.GetExamplesRepositoryURI()
-	kogitoApp.Spec.Build.GitSource.ContextDir = kogitoAppDeployment.ContextDir
-	kogitoApp.Spec.Build.GitSource.Reference = config.GetExamplesRepositoryRef()
+
+	if gitSourceURI := kogitoAppDeployment.GitSourceURI; len(gitSourceURI) > 0 {
+		kogitoApp.Spec.Build.GitSource.URI = gitSourceURI
+		kogitoApp.Spec.Build.GitSource.ContextDir = kogitoAppDeployment.ContextDir
+
+		if ref := kogitoAppDeployment.GitSourceReference; len(ref) > 0 {
```

Comment:
`kogitoApp.Spec.Build.GitSource.Reference` is with `omitempty` configuration so don't need to do this check

Reply:
Actually I need to do the check.
Because reference can be empty, therefore default value is applied - empty string.

Reply:
weird... I thought it would be ignored

Reply:
it can't be ignored as it is not a pointer but structure field
omitempty serves as a hint for marshaller to not marshall the field if it contains default value

Reply:
now I get it :)

Reply:
@sutaakar please implement a function on `KogitoAppSpec` like `IsGitSourceEmpty` and do the checks there. We can reuse it and the code will be clearer.